samplers/src/test/java/internal/CelBasedSamplerComponentProviderTest.java

```diff
@@ -0,0 +1,246 @@
+                + "expressions:\n"
+                + "  - action: DROP\n"
+                + "    expression: 'attribute[\"http.request.method\"] == \"GET\"'\n"
+                + "  - action: DROP\n"
+                + "    expression: 'attribute[\"url.path\"] == \"/foo/bar\"'\n",
+            dropSampler(
+                Sampler.alwaysOff(),
+                "attribute[\"http.request.method\"] == \"GET\"",
+                "attribute[\"url.path\"] == \"/foo/bar\"")));
+  }
+
+  @ParameterizedTest
+  @MethodSource("createInvalidArgs")
+  void create_Invalid(String yaml, String expectedErrorMessage) {
+    DeclarativeConfigProperties configProperties =
+        DeclarativeConfiguration.toConfigProperties(
+            new ByteArrayInputStream(yaml.getBytes(StandardCharsets.UTF_8)));
+
+    assertThatThrownBy(() -> PROVIDER.create(configProperties))
+        .isInstanceOf(DeclarativeConfigException.class)
+        .hasMessage(expectedErrorMessage);
+  }
+
+  static Stream<Arguments> createInvalidArgs() {
+    return Stream.of(
+        Arguments.of(
+            "expressions:\n" + "  - action: DROP\n" + "    expression: 'spanKind == \"CLIENT\"'\n",
+            "cel_based sampler .fallback_sampler is required but is null"),
+        Arguments.of(
+            "fallback_sampler:\n"
+                + "  foo:\n"
+                + "expressions:\n"
+                + "  - action: DROP\n"
+                + "    expression: 'spanKind == \"CLIENT\"'\n",
+            "cel_based sampler failed to create .fallback_sampler sampler"),
+        Arguments.of(
+            "fallback_sampler:\n" + "  always_on: {}\n",
+            "cel_based sampler .expressions is required"),
+        Arguments.of(
+            "fallback_sampler:\n" + "  always_on: {}\n" + "expressions: []\n",
+            "cel_based sampler .expressions is required"),
+        Arguments.of(
+            "fallback_sampler:\n" + "  always_on: {}\n" + "expressions:\n" + "  - action: DROP\n",
+            "cel_based sampler .expressions[].expression is required"),
+        Arguments.of(
+            "fallback_sampler:\n"
+                + "  always_on: {}\n"
+                + "expressions:\n"
+                + "  - expression: 'spanKind == \"CLIENT\"'\n",
+            "cel_based sampler .expressions[].action is required"),
+        Arguments.of(
+            "fallback_sampler:\n"
+                + "  always_on: {}\n"
+                + "expressions:\n"
+                + "  - action: INVALID\n"
+                + "    expression: 'spanKind == \"CLIENT\"'\n",
+            "cel_based sampler .expressions[].action must be RECORD_AND_SAMPLE or DROP"),
+        Arguments.of(
+            "fallback_sampler:\n"
+                + "  always_on: {}\n"
+                + "expressions:\n"
+                + "  - action: DROP\n"
+                + "    expression: 'invalid cel expression!'\n",
+            "Failed to compile CEL expression: invalid cel expression!"));
+  }
```

Comment:
Maybe I missed it, but it would be cool to add coverage for the case where `expressions` exists but is empty.

Reply:
A new test case was added for this case. It will result in a similar error as the CEL expression is invalid. I opted for a compile error from the CEL engine rather than an is empty check.